packages/babel-core/README.md

```diff
@@ -123,6 +123,78 @@ const parsedAst = babylon.parse(sourceCode, { allowReturnOutsideFunction: true }
 const { code, map, ast } = babel.transformFromAstSync(parsedAst, sourceCode, options);
 ```
 
+## babel.parse(code: string, [options?](#options): Object)
+
+Given some code, parse it using Babel's standard behavior. Referenced presets and
+plugins will be loaded such that optional syntax plugins are automatically
+enabled.
+
+
+## Advanced APIs
+
+Many systems that wrap Babel like to automatically inject plugins and presets,
+or override options. To accomplish this goal, Babel exposes several functions
+that aid in loading the configuration part-way without transforming.
+
+### babel.loadOptions([options?](#options): Object)
+
+Resolve Babel's options fully, resulting in an options object where:
+
+* `opts.plugins` is a full list of `Plugin` instances.
+* `opts.presets` is empty and all presets are flattened into `opts`.
+* It can be safely passed back to Babel. Fields like `babelrc` have been set to 
+  false so that later calls to Babel will not make a second attempt to load
+  config files.
+
+`Plugin` instances aren't meant to be manipulated directly, but often
+callers will serialize this `opts` to JSON to use it as a cache key representing
+the options Babel has received. Caching on this isn't 100% guaranteed to
+invalidate properly, but it is the best we have at the moment.
+
+
+### babel.loadPartialConfig([options?](#options): Object): PartialConfig
+
+To allow systems to easily manipulate and validate a user's config, this function
+resolves the plugins and presets and proceeds no further. The expectation is
+that callers will take the config's `.options`, manipulate it as then see fit
+and pass it back to Babel again.
+
+* `babelrc: string | void` - The path of the `.babelrc` file, if there was one.
```

Comment:
Does `loadPartialConfig` only load the first `.babelrc` or it also loads its parents (when needed)? Also, what about `babel.config.js`?

Reply:
It basically does everything to know what the configured plugins and presets will be, without executing them, so that includes loading all config files, and any options passed in programmatically, and merging them all together into a single config object. It just skips actually calling the plugin/presets function, and flattening the presets themselves.